src/main.rs

```diff
@@ -0,0 +1,253 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#[macro_use(value_t)]
+extern crate clap;
+
+use clap::{App, AppSettings, Arg, SubCommand};
+use manticore::io::StdWrite;
+use manticore::protocol;
+use manticore::protocol::firmware_version;
+use manticore::protocol::wire::FromWire;
+use manticore::protocol::wire::ToWire;
+use manticore::protocol::CommandType;
+use manticore::protocol::Header;
+use serde_json;
+use std::fs::OpenOptions;
+use std::io::prelude::*;
+use std::io::BufReader;
+
+/// Deserializes a message in JSON format from `reader` and then serializes the message in wire
+/// format to `writer`.
+fn from_json_to_wire<'a, T, R, W>(reader: R, writer: W)
+where
+    T: ToWire + serde::de::Deserialize<'a>,
```

Comment:
Can you import `Deserialize` directly? I don't think it's ambiguoous. (I suggest keeping `serde_json::Deserializer`, though, since that name is a bit more unclear.

Reply:
Done.